src/js/dom.js

```diff
@@ -0,0 +1,181 @@
+    messageText.classList.add('success-message');
+  } else if (type === 'error') {
+    messageText.classList.remove('success-message');
+    messageText.classList.add('error-message');
+  }
+  hideMessageBox(messageText);
+};
+
+export const renderComments = (commentsArray) => {
+  if (commentsArray && commentsArray.length > 0) {
+    const commentsCounterDiv = modalContentContainer.childNodes[9].childNodes[1];
+    const commentsSection = modalContentContainer.childNodes[9].childNodes[3];
+    commentsSection.innerHTML = '';
+    commentsCounterDiv.innerHTML = `Comments (${commentsArray.length})`;
+
+    commentsArray.forEach((object) => {
+      commentsSection.innerHTML += `<div class="flex flex-wrap justify-start">
+          <div class="flex items-center mb-1">
+            <p class="bg-blue-200 text-green-600 py-1 px-2 font-bold flex flex-wrap">${object.creation_date.replaceAll('-', '/')}</p>
+            <p class="bg-blue-200 text-blue-500 py-1 px-2 font-bold rounded-tr-full rounded-br-full flex flex-wrap">${object.username}</p>
+          </div>
+          <p class="py-1 px-4 font-normal text-justify">${object.comment}</p>
+        </div>`;
+    });
+  }
+};
+
+const renderCommentForm = () => `
+  <div class="my-5 md:my-8 pt-2 border-t-2 border-gray-200">
+    <h4 class="mt-5 text-lg md:text-2xl leading-6 font-medium text-gray-500">
+      Add a comment
+    </h4>
+    <form action="#" method="POST" class="flex flex-col px-8 md:px-10 lg:px-48 xl:px-64 md:pt-5">
+      <p id="alertMessage" class="message-box"></p>
+      <input type="text" id="commenterName" placeholder="Your name" class="p-3 my-3 rounded-md shadow-3xl border border-2 border-gray-300 focus:outline-none" />
+      <textarea type="text" id="commenterMessage" placeholder="Your insights" class="p-3 my-3 rounded-md shadow-3xl border border-2 border-gray-300 focus:outline-none"></textarea>
+      <button class="bg-transparent focus:outline-none hover:text-white text-gray-500 w-32 h-10 rounded-tr-full rounded-bl-full shadow-3xl my-3 submit-comment" type="button">Comment</button>
+    </form>
+  </div>
+`;
+
+const resetInputFields = (args) => {
+  args.forEach((arg) => {
+    arg.value = '';
+  });
+};
+
+const submitComment = (newsId) => {
+  const form = document.querySelector('form');
+  const commenterName = form.childNodes[3];
+  const commenterMessage = form.childNodes[5];
+  const formButton = form.childNodes[7];
+
+  formButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+    if (commenterName.value && commenterMessage.value !== '') {
+      const newComment = {
+        item_id: newsId.toString(),
+        username: commenterName.value,
+        comment: commenterMessage.value,
+      };
+
+      await postComment(newComment);
+      getComments(newsId.toString());
+      resetInputFields([commenterName, commenterMessage]);
+    } else {
+      showMessage('All input fields are required', 'error');
+    }
+  });
+};
```

Comment:
- [ ] This `submitComment` function has a bug and it happens when you try to post a comment by pushing the `Enter` key and without filling the `comment` field. The event is triggered because you are using a `form` and there are two issues here. First, it is not the correct element that triggers the event and secondly, the event attached to your listener is not the good one. Notice that the data sent by a `form` are sent through the `submit` event and that event is triggered by the `form element` itself, that's the reason why you can use the `e.preventDefault` function to handle the form default behavior.

![Screenshot from 2021-10-15 16-32-54](https://user-images.githubusercontent.com/40503704/137513995-3745b37d-79da-4eb2-9187-fac9bfc37cfc.png)

- [ ] Here are some hints to fix this
- Attach the `form` element instead of the `formButton` to the listener 
- Change the event from `click` to `submit`